flink-runtime/src/main/java/org/apache/flink/runtime/messages/webmonitor/JobDetails.java

```diff
@@ -84,10 +85,13 @@ public class JobDetails implements Serializable {
     /**
      * The map holds the attempt number of the current execution attempt in the Execution, which is
      * considered as the representing execution for the subtask of the vertex. The keys and values
-     * are JobVertexID -> SubtaskIndex -> CurrentExecutionAttemptNumber. It is used to accumulate
-     * the metrics of a subtask in MetricFetcher.
+     * are JobVertexID -> SubtaskIndex -> CurrentExecutionAttemptNumbers. The first element in the
+     * CurrentExecutionAttemptNumbers is of the representative attempt.
+     *
+     * <p>The field is excluded from the json. Any usage from the web UI and the history server is
+     * not allowed.
      */
-    private final Map<String, Map<Integer, Integer>> currentExecutionAttempts;
+    private final Map<String, Map<Integer, List<Integer>>> currentExecutionAttempts;
```

Comment:
We may change the List to an explicit structure. It could be defined inside the `JobDetails`

It might be sth like

```java
    public static final class CurrentAttempts {
        public final int representativeAttempts;

        public final List<Integer> currentAttempts;

        public CurrentAttempts(int representativeAttempts, List<Integer> currentAttempts) {
            this.representativeAttempts = representativeAttempts;
            this.currentAttempts = Collections.unmodifiableList(currentAttempts);
        }
    }
```